rules/mastg-android-object-deserialization.yml

```
@@ -8,5 +8,6 @@ rules:
     message: "[MASVS-CODE-4] The application make use of Object deserialization in the code."
     patterns:
     - pattern-either:
-      - pattern: import java.io.Serializable
-      - pattern: class $CLASS implements Serializable
\ No newline at end of file
+        - pattern:
+            $VAR = new java.io.ObjectInputStream(...);
+            $OBJ = $VAR.readObject();
```

```suggestion
        - pattern:
            $VAR = new java.io.ObjectInputStream(...);
            $OBJ = $VAR.readObject();
```
patter neither is not need here you can just use patter directly
```
rules:
  - id: mastg-android-object-deserialization
    severity: WARNING
    languages:
      - java
    metadata:
      summary: This rule looks for use of Object Serialization
    message: "[MASVS-CODE-4] The application make use of Object deserialization in the code."
    pattern: |
      $VAR = new java.io.ObjectInputStream(...);
      $OBJ = $VAR.readObject();
```
Use this and re-run the semgrep